crates/cxx-qt-lib/src/core/qtlogging.cpp

```diff
@@ -0,0 +1,68 @@
+// clang-format off
+// SPDX-FileCopyrightText: 2024 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// clang-format on
+// SPDX-FileContributor: Joshua Goins <joshua.goins@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+#include "cxx-qt-lib/qtlogging.h"
+
+#include "../assertion_utils.h"
+
+// QMessageLogContext has three "const char*" members for line, category, etc
+// https://codebrowser.dev/qt5/qtbase/src/corelib/global/qlogging.h.html#QMessageLogContext
+assert_alignment_and_size(QMessageLogContext,
+                          alignof(intptr_t),
+                          sizeof(intptr_t) * 4);
+
+static_assert(::std::is_trivially_copyable<QMessageLogContext>::value,
+              "QMessageLogContext must be trivially copyable");
```

Comment:
so looks like the CI failure on Windows at least might be coming from this?

```
  qstring.cpp
  src/core/qstring.cpp(46): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
  qstringlist.cpp
  qurl.cpp
  qtlogging.cpp
  src/core/qtlogging.cpp(17): error C2338: static_assert failed: 'QMessageLogContext must be trivially copyable'
```

I also note that qstring warning that is unrelated but interesting :-)

Reply:
Meh I somehow missed the Q_DISABLE_COPY on the struct

Reply:
Fixed